discord/ext/commands/converter.py

```diff
@@ -159,6 +159,65 @@ class UserConverter(IDConverter):
 
         return result
 
+class MessageConverter(IDConverter):
+    """Converts to a :class:`discord.Message`.
+
+    The lookup strategy is as follows (in order):
+
+    1. Lookup by "{channel ID}-{message ID}" (got by shift-clicking on "Copy ID")
+    2. Lookup by message ID (the message **must** be in the context channel)
+    3. Lookup by message URL
+    """
+
+    async def convert(self, ctx, argument):
+        prog = re.compile(r'^(?:(?P<channel_id>[0-9]{15,21})-)?(?P<message_id>[0-9]{15,21})$')
+        match = prog.match(argument)
+        if match:
+            message_id = int(match.group("message_id"))
+            message = ctx.bot._connection._get_message(message_id)
+            if message:
+                return message
+            channel_id = match.group("channel_id")
```

Comment:
```suggestion
            channel_id = match.group("channel_id")
            channel = ctx.bot.get_channel(int(channel_id)) or ctx.channel
```

Do this instead of duplicating all that code down there.

Reply:
Using `int()` would raise `TypeError` if there's no channel ID.
```py
>>> import re
>>> id_regex = re.compile(r'^(?:(?P<channel_id>[0-9]{15,21})-)?(?P<message_id>[0-9]{15,21})$')
>>> match = id_regex.match("987654321012345")
>>> int(match.group("channel_id")) or 42
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```